src/goto-instrument/contracts/contracts.cpp

```diff
@@ -598,22 +672,54 @@ bool code_contractst::apply_function_contract(
 }
 
 void code_contractst::apply_loop_contract(
-  const irep_idt &function,
+  const irep_idt &function_name,
   goto_functionst::goto_functiont &goto_function)
 {
   local_may_aliast local_may_alias(goto_function);
   natural_loops_mutablet natural_loops(goto_function.body);
 
-  // Iterate over the (natural) loops in the function,
-  // and apply any invariant annotations that we find.
+  struct loop_graph_nodet : public graph_nodet<empty_edget>
+  {
+    typedef const goto_programt::targett &targett;
+    typedef const typename natural_loops_mutablet::loopt &loopt;
+
+    targett target;
+    loopt loop;
+
+    loop_graph_nodet(targett t, loopt l) : target(t), loop(l)
+    {
+    }
+  };
+  grapht<loop_graph_nodet> loop_nesting_graph;
```

Comment:
Why do we need to keep this struct here and not on a separate header file?

Reply:
It seemed too tiny and specific to this use case and not very useful generally, but I am open to moving it elsewhere.

@tautschnig, what do you suggest?

Reply:
It would be really helpful to have documentation that answers the "why" question of why this entire struct exists in the first place. That might help answer whether here or elsewhere is the right place.

Reply:
This looks self-explanatory to me: we are creating a graph node that stores a loop information. Do you have any suggestions on what you'd like to see in a comment?

Reply:
I added a couple of suggestions on my previous comment.